packages/notifi-react-card/lib/components/subscription/EventTypeFusionHealthCheckRow.tsx

```diff
@@ -219,42 +246,41 @@ export const EventTypeFusionHealthCheckRow: React.FC<
   );
 
   const handleCustomRatioButtonNewSubscription = () => {
-    if (loading || isNotificationLoading) {
+    if (loading || isNotificationLoading || !customInputRef.current) {
       return;
     }
-
+    const inputValue = customInputRef.current.value;
     let regex = new RegExp(/^[0-9.]+$/);
+    let inputNumber = null;
     switch (config.numberType) {
       case 'percentage':
         regex = new RegExp(/^[0-9.%]+$/);
+        inputNumber = getParsedPercentage(inputValue);
         break;
       case 'price':
         regex = new RegExp(/^[0-9.$]+$/);
+        inputNumber = getParsedPrice(inputValue);
         break;
+      default: // 'integer'
+        inputNumber = parseFloat(inputValue);
     }
 
-    if (!customInputRef.current || !regex.test(customInputRef.current.value)) {
-      return setErrorMessage(INVALID_NUMBER);
+    const { isInputValid, inputInvalidWarning } = validateInput(
+      inputValue,
+      regex,
+      inputNumber,
+    );
+
+    if (!isInputValid) {
+      return setErrorMessage(inputInvalidWarning);
     }
 
     setErrorMessage('');
     setIsNotificationLoading(true);
 
     customInputRef.current.placeholder = 'Custom';
-    let ratioNumber = null;
-    switch (config.numberType) {
-      case 'percentage':
-        ratioNumber = getParsedPercentage(customInputRef.current.value);
-        break;
-      case 'price':
-        ratioNumber = getParsedPrice(customInputRef.current.value);
-        break;
-      default: // 'integer'
-        ratioNumber = parseFloat(customInputRef.current.value);
-    }
-
-    if (ratioNumber && ratioNumber >= 0 && customValue) {
-      subscribeAlert({ eventType: config, inputs }, ratioNumber)
+    if (inputNumber && inputNumber >= 0 && customValue) {
```

Comment:
why do we need to check `inputNumber >= 0`?

Reply:
Right, we should not restrict users not to subscribe 0 since we now able to input valid number range.